src/hooks/useAutoSlippageTolerance.ts

```diff
@@ -14,14 +14,28 @@ import useStablecoinPrice, { useStablecoinValue } from './useStablecoinPrice'
 const V3_SWAP_DEFAULT_SLIPPAGE = new Percent(50, 10_000) // .50%
 const ONE_TENTHS_PERCENT = new Percent(10, 10_000) // .10%
 export const DEFAULT_AUTO_SLIPPAGE = ONE_TENTHS_PERCENT
+const GAS_ESTIMATE_BUFFER = new Percent(10, 100) // 10%
 
 /**
  * Return a guess of the gas cost used in computing slippage tolerance for a given trade
  * @param trade the trade for which to _guess_ the amount of gas it would cost to execute
  */
 function guesstimateGas(trade: Trade<Currency, Currency, TradeType> | undefined): number | undefined {
   if (!!trade) {
-    return 100_000 + trade.swaps.reduce((memo, swap) => swap.route.pools.length + memo, 0) * 30_000
+    let gas = 0
+    for (const { route } of trade.swaps) {
+      if (route.protocol === Protocol.V2) {
+        gas += 135000 + route.pools.length * 50_000
+      } else if (route.protocol === Protocol.V3) {
+        // V3 gas costs scale on initialized ticks being crossed, but we don't have that data here.
+        // We bake in some tick crossings into the base 100k cost.
+        gas += 100_000 + route.pools.length * 70_000
+      } else {
```

Comment:
does this mean it's `v1`?

Reply:
No interleaved routes are a new feature we are launching in the next couple weeks. It means the route can now be like:

ETH --V2pool --> USDC --V3pool--> DAI

I'll be letting Eric know that he needs to update this code when he does the PR for it

Reply:
ahh i see, thanks!